src/fluent/watch.ts

```diff
@@ -361,17 +377,43 @@ export class Watcher<T extends GenericClass> {
 
       // Build the URL and request options
       const { opts, url } = await this.#buildURL(true, this.#resourceVersion);
+      let agentOptions;
+      if (opts.agent && opts.agent instanceof https.Agent) {
+        agentOptions = {
+          key: opts.agent.options.key,
+          cert: opts.agent.options.cert,
+          ca: opts.agent.options.ca,
+          rejectUnauthorized: false,
+        };
+      }
 
-      // Create a stream to read the response body
-      this.#stream = byline.createStream();
-
-      // Bind the stream events
-      this.#stream.on("error", this.#errHandler);
-      this.#stream.on("close", this.#streamCleanup);
-      this.#stream.on("finish", this.#streamCleanup);
+      const agent = new Agent({
+        // https://github.com/nodejs/undici/blob/87d7ccf6b51c61a4f4a056f7c2cac78347618486/docs/docs/api/Errors.md?plain=1#L16
+        // https://github.com/nodejs/undici/blob/87d7ccf6b51c61a4f4a056f7c2cac78347618486/docs/docs/api/Client.md?plain=1#L24
+        keepAliveMaxTimeout: 600000,
+        keepAliveTimeout: 600000,
+        bodyTimeout: 0,
+        connect: {
+          ca: agentOptions?.ca,
+          cert: agentOptions?.cert,
+          key: agentOptions?.key,
+        },
+      });
+
+      const token = await this.#getToken();
```

Comment:
still struggling with this and think we need to dig into what changed/how to adapt the wiring in buildURL vs doing this

Reply:
Yeah, for sure we can change buildURL, i didn't want to touch it in this bc because I didn't want to change change the list or existing fetch call that list uses bc the work would have snowballed and I just wanted to check and see if the watch would actually be improved. All of that stuff is definitely on the table IMO

Reply:
If this were the direction that we go, we can probably obtain this info from kubernetes/client-node 

https://github.com/kubernetes-client/javascript/blob/82c85b94290458385bf976ad9b801702953e11e1/src/config.ts#L216